src/disk_io_thread.cpp

```diff
@@ -2084,22 +2086,27 @@ namespace libtorrent
 		DLOG("kick_hasher: %d - %d (piece: %d offset: %d)\n"
```

Comment:
And to answer you question, yes, only one thread should be kicking the hasher at a time.